handler/transport.go

```diff
@@ -33,8 +33,10 @@ type transport struct {
 }
 
 func (t *transport) RoundTrip(r *http.Request) (*http.Response, error) {
-	if err := t.prov.VerifyRoleToken(r.Context(), r.Header.Get(t.cfg.RoleHeader), r.Method, r.URL.Path); err != nil {
-		return nil, errors.Wrap(err, ErrMsgVerifyRoleToken)
+	if t.cfg.BypassUrlPath == "" || t.cfg.BypassUrlPath != r.URL.Path {
```

Comment:
I think we should change the bypass logic here for the performance, 
https://github.com/yahoojapan/authorization-proxy/blob/42153e3ae294b928ef6627dce7ce96b5b3de5447/handler/handler.go#L62

or we should use the function pointer to change the behavior of the bypass logic.

Reply:
What is the performance concern you have? I think we need to check all the requests.

transport.go already wrapped the default http.Transport.RoundTrip function.

Reply:
POC: https://github.com/yahoojapan/authorization-proxy/pull/42

Reply:
what do you think? which one is better? I think the POC one is the better implementation

Reply:
no preferences, maybe do performance test later

Reply:
https://github.com/yahoojapan/authorization-proxy/blob/1360c02583f954790106fb51f367723d4d034239/handler/transport.go#L37-L39

the for loop will check for empty anyway, we don't need this PR https://github.com/yahoojapan/authorization-proxy/pull/42?
Or, better to separate the object?

Reply:
spec changed, code not valid anymore